src/base/Memory.hh

```diff
@@ -3,17 +3,15 @@
 // See the top-level COPYRIGHT file for details.
 // SPDX-License-Identifier: (Apache-2.0 OR MIT)
 //---------------------------------------------------------------------------//
-//! \file Utils.hh
+//! \file Memory.hh
 //---------------------------------------------------------------------------//
 #pragma once
 
-#include "../Span.hh"
-#include "../Types.hh"
+#include "Span.hh"
+#include "Types.hh"
 
 namespace celeritas
 {
-namespace detail
```

Comment:
Do we want these free functions directly in the `celeritas` namespace?

Reply:
Following Google best practices, I want to avoid having sub-namespaces other than `detail`, and I'm not a fan of the pattern of grouping static methods inside a class. Did you have something in mind?

Reply:
No, that is find by me - the code base probably won't get big enough to overly pollute `celeritas`.  The only other option would be to break out by package.  However, since this is in a detail directory I thought that it should retain the `detail` sub-namespace.

Reply:
Oh, I see what you mean -- for some reason it's showing up under the old filename, but in this MR I moved it to `src/base/Memory.hh`.